src/TagsInput/TagsInput.js

```diff
@@ -11,156 +16,154 @@ import InputField from './components/InputField';
+
+const boxShadow = isFocus => {
+  if (Platform.OS === 'web') {
+    return {
+      boxShadow: isFocus
+        ? `${defaultTokens.borderColorInputFocus} 0 0 0 2px inset`
+        : `${defaultTokens.borderColorInput} 0 0 0 1px inset`,
+    };
+  }
+  return null;
 };
 
 export default class TagInput extends React.Component<Props, State> {
-  inputRef: ?{ current: null | React.Element<typeof TextInput> };
+  inputRef: any;
+  scrollRef: any;
+
   static defaultProps = {
-    fontSize: 16,
-    autoFocus: true,
+    fontSize: parseFloat(defaultTokens.fontSizeButtonLarge),
+    tags: [],
   };
 
   constructor(props: Props) {
     super(props);
     this.inputRef = React.createRef();
+    this.scrollRef = React.createRef();
 
     this.state = {
-      value: props.value ?? '',
-      containerWidth: null,
-      inputWidth: null,
-      isFocus: props.autoFocus,
+      value: props.value || '',
+      isFocus: false,
     };
   }
 
-  componentDidUpdate = (prevProps: Props) => {
-    const { value } = this.props;
-    if (value !== prevProps.value) {
-      this.setState({ value });
-    }
-  };
-
-  setContainerWidth = (event: ViewLayoutEvent) => {
-    this.setState({ containerWidth: event.nativeEvent.layout.width });
-  };
-
-  setInputWidth = (event: ViewLayoutEvent) => {
-    this.setState({ inputWidth: event.nativeEvent.layout.width });
-  };
-
-  subtractFromContainerWidth = (width: ?number) => {
-    const { containerWidth } = this.state;
-    if (containerWidth && width) {
-      const newWidth = containerWidth - width;
-      return newWidth < 0 ? 'auto' : newWidth;
+  static getDerivedStateFromProps(props: Props, state: State) {
+    // if the "props.value" has changed update state
+    const { value } = props;
+    if (typeof value === 'string' && value !== state.value) {
+      return {
+        value,
+      };
     }
     return null;
-  };
+  }
 
   getPlaceholder = () => {
-    const { selected, placeholder } = this.props;
-    return selected?.length === 0 && placeholder ? placeholder : null;
+    const { tags, placeholder } = this.props;
+    // Don't render placeholder if the Tags are rendered.
+    return tags.length === 0 && placeholder ? placeholder : null;
   };
 
   handleChange = (value: string) => {
     const { value: oldValue } = this.state;
     const { onChangeText } = this.props;
     if (value !== oldValue) {
-      this.setState({ value });
+      this.setState({ value }, this.scrollRef.current.scrollToEnd);
       onChangeText(value);
     }
   };
 
-  handleClear = () => {
-    const { onClearPress } = this.props;
-    this.setState({ value: '' });
-
-    // $FlowFixMe property focus is missing in object type
-    Platform.OS === 'web' && this.inputRef.current.focus();
-
-    onClearPress?.();
-  };
-
   handleOnFocus = () => {
+    const { isFocus } = this.state;
+    if (isFocus) return;
     this.setState({ isFocus: true });
+    this.inputRef.current && this.inputRef.current.focus();
```

Comment:
```suggestion
    this.inputRef.current?.focus.?();
```
or something like that?

Reply:
And maybe you could add it as callback after you set the state.

Reply:
I will consider the callback.

About the `optional-chaining`, I tried it before, but flow starts to scream on me this:
`Flow does not yet support method or property calls in optional chains.`